management_app/views/faculty.py

```diff
@@ -1,42 +1,221 @@
-from flask import Flask, Blueprint, send_from_directory, render_template, request, redirect, url_for
+import pandas as pd
+import traceback
+from flask import Blueprint, render_template, request, redirect, url_for
 from werkzeug.utils import secure_filename
 
+from management_app.db import get_db
 from management_app.views.auth import login_required
-from management_app.views.utils import download_file, upload_file, remove_upload_file
+from management_app.views.utils import download_file, upload_file, remove_upload_file, get_upload_filepath
 
 faculty = Blueprint('faculty', __name__, url_prefix='/faculty')
 
-@faculty.route('/')
+@faculty.route('/', methods=['GET', 'POST'])
 @login_required
 def index():
-    # TODO: Get data from database
-    faculties = [
-        {'name': 'Professor Mark', 'email': 'mark@uci.edu', 'role': 'Tenured Faculty', 'required_point': 3.5,
-         'prev_balance': 0.1250, 'estimated_balance': 0.2500},
-        {'name': 'Professor 2', 'email': 'mark1@uci.edu', 'role': 'Pre Tenured Faculty 1 st', 'required_point': 2.5,
-         'prev_balance': 0.1000, 'estimated_balance': 0.1250},
-        {'name': 'Professor 3', 'email': 'mark3@uci.edu', 'role': 'PoT Tenured Faculty', 'required_point': 6.5,
-         'prev_balance': 1.4375, 'estimated_balance': 0.9375}
-    ]
-    return render_template('faculty/index.html', faculties=faculties)
-
-
-@faculty.route('/data-templates/<filename>')
+    years = ['2020-2021', '2019-2020']
```

Comment:
Yes, I'll keep in mind this will be a dynamic input but I'll figure out how to make it later. For this PR, let's leave it as hard-coded at this moment.